src/Ilios/CoreBundle/Classes/CourseRollover.php

```diff
@@ -0,0 +1,399 @@
+    {
+        /* @var CourseLearningMaterialInterface[] $originalCourseLearningMaterials */
+        $originalCourseLearningMaterials
+            = $this->courseLearningMaterialManager->findBy(['course'=>$originalCourse]);
+
+        foreach ($originalCourseLearningMaterials as $originalCourseLearningMaterial) {
+            /* @var CourseLearningMaterialInterface $newCourseLearningMaterial */
+            $newCourseLearningMaterial = $this->courseLearningMaterialManager->create();
+            $newCourseLearningMaterial->setNotes($originalCourseLearningMaterial->getNotes());
+            $newCourseLearningMaterial->setRequired($originalCourseLearningMaterial->isRequired());
+            $newCourseLearningMaterial->setPublicNotes($originalCourseLearningMaterial->hasPublicNotes());
+            $newCourseLearningMaterial->setCourse($newCourse);
+            $newCourseLearningMaterial->setLearningMaterial($originalCourseLearningMaterial->getLearningMaterial());
+            $newCourseLearningMaterial->setMeshDescriptors($originalCourseLearningMaterial->getMeshDescriptors());
+            
+            $this->courseLearningMaterialManager->update($newCourseLearningMaterial, false, false);
+        }
+    }
+
+    /**
+     * @param CourseInterface $newCourse
+     * @param CourseInterface $originalCourse
+     * @param array $options
+     * @param int $offsetInWeeks
+     */
+    protected function rolloverSessions(
+        CourseInterface $newCourse,
+        CourseInterface $originalCourse,
+        $options,
+        $offsetInWeeks
+    ) {
+        /* @var SessionInterface[] $originalCourseSessions */
+        $originalCourseSessions = $this->sessionManager->findBy(['course'=>$originalCourse]);
+
+        foreach ($originalCourseSessions as $originalCourseSession) {
+            /* @var SessionInterface $newSession */
+            $newSession = $this->sessionManager->create();
+            $newSession->setCourse($newCourse);
+            $newSession->setTitle($originalCourseSession->getTitle());
+            $newSession->setAttireRequired($originalCourseSession->isAttireRequired());
+            $newSession->setEquipmentRequired($originalCourseSession->isEquipmentRequired());
+            $newSession->setSessionType($originalCourseSession->getSessionType());
+            $newSession->setSupplemental($originalCourseSession->isSupplemental());
+            $newSession->setPublishedAsTbd(0);
+            $newSession->setPublished(0);
+
+            //SESSION LEARNING MATERIALS
+            if (empty($options['skip-session-learning-materials'])) {
+                $this->rolloverSessionLearningMaterials($newSession, $originalCourseSession);
+            }
+
+            //SESSION OBJECTIVES
+            if (empty($options['skip-session-objectives'])) {
+                $newSession->setObjectives($originalCourseSession->getObjectives());
```

Comment:
This will re-link the same objective to the new session.  Is that what we have been doing?  Or did we usually copy the objective text into the new session?  I'm worried about creating links which, when edited, would then edit the original session's objectives.